versions-common/src/main/java/org/codehaus/mojo/versions/utils/PropertyComparator.java

```diff
@@ -42,14 +41,18 @@ public enum PropertyComparator implements Comparator<Property>
      * @see java.util.Comparator#compare(Object, Object)
      * @since 1.0-beta-1
      */
+    @SuppressWarnings( "checkstyle:InnerAssignment" )
     public int compare( Property p1, Property p2 )
     {
+        int r;
         return p1 == p2
                 ? 0
                 : p1 == null
-                    ? 1
-                    : p2 == null
-                        ? -1
-                        : StringUtils.compare( p1.getName(), p2.getName() );
+                        ? 1
+                        : p2 == null
+                                ? -1
+                                : ( r = StringUtils.compare( p1.getName(), p2.getName() ) ) == 0
+                                        ? StringUtils.compare( p1.getValue(), p2.getValue() )
```

Comment:
The original idea behind omitting the value in comparison was that properties with different values were treated as the same property. Was there a reason you wanted to treat properties with different values as different properties?

Reply:
The reason is that a child may define its own property value. If comparison is done only by name, then only one property entry will be in the final report and not all, that are possible.